src/transformers/image_processing_utils_fast.py

```diff
@@ -131,3 +121,882 @@ def get_max_height_width(images: List["torch.Tensor"]) -> Tuple[int]:
+        )
+        size = get_size_dict(size=size, default_to_square=default_to_square) if size is not None else None
+        resample = resample if resample is not None else self.resample
+        do_center_crop = do_center_crop if do_center_crop is not None else self.do_center_crop
+        crop_size = crop_size if crop_size is not None else self.crop_size
+        crop_size = get_size_dict(crop_size, param_name="crop_size") if crop_size is not None else None
+        do_rescale = do_rescale if do_rescale is not None else self.do_rescale
+        rescale_factor = rescale_factor if rescale_factor is not None else self.rescale_factor
+        do_normalize = do_normalize if do_normalize is not None else self.do_normalize
+        image_mean = image_mean if image_mean is not None else self.image_mean
+        image_std = image_std if image_std is not None else self.image_std
+        do_convert_rgb = do_convert_rgb if do_convert_rgb is not None else self.do_convert_rgb
+        return_tensors = "pt" if return_tensors is None else return_tensors
+        device = kwargs.pop("device", None)
+
+        images, image_mean, image_std, size, crop_size, interpolation = self.prepare_process_arguments(
+            images=images,
+            do_resize=do_resize,
+            size=size,
+            resample=resample,
+            do_center_crop=do_center_crop,
+            crop_size=crop_size,
+            do_rescale=do_rescale,
+            rescale_factor=rescale_factor,
+            do_normalize=do_normalize,
+            image_mean=image_mean,
+            image_std=image_std,
+            do_convert_rgb=do_convert_rgb,
+            return_tensors=return_tensors,
+            data_format=data_format,
+            input_data_format=input_data_format,
+            device=device,
+            **kwargs,
+        )
+
+        processed_images = []
+        for image in images:
+            if do_resize:
+                image = self.resize(
+                    image=image,
+                    size=size,
+                    resample=interpolation,
+                )
+
+            if do_center_crop:
+                image = self.center_crop(image, crop_size)
+
+            if do_rescale and do_normalize:
+                # fused rescale and normalize
+                image = self.normalize(image.to(dtype=torch.float32), image_mean, image_std)
+            elif do_rescale:
+                image = image * rescale_factor
+            elif do_normalize:
+                image = self.normalize(image, image_mean, image_std)
+
+            processed_images.append(image)
+        images = processed_images
+
+        return BatchFeature(data={"pixel_values": torch.stack(images, dim=0)}, tensor_type=return_tensors)
+
+    def to_dict(self):
+        encoder_dict = super().to_dict()
+        encoder_dict.pop("_valid_processor_keys", None)
+        return encoder_dict
+
+
+class LlavaPatchingMixin:
```

Comment:
The patching might be different for different models, so we might end up adding more and more such kind of mixins to common.. I think this should live a particular model code with `# Copied from`. 
Alternatively, we might have here an interface with abstract methods defined and override them for particular models (but might be not that readable).

Reply:
Yes I think I agree with this, especially since this is common to only two models (for now). 
Do you think having mixins at all for some common preprocessing operations is a good idea? This idea came from the fact that image processors similar to the DETR one have more than a thousand lines where most if not all functions are copied from, and there are 7 or so DETR like image processors.

Reply:
It might be a `PatchingMixin` which will wpar usual preprocessing to apply it over patches, however patching strategy, should be implemented separately because it's different. Otherwise, we will have a lot of copies of this class with different strategies.

Reply:
+1 on this